pkg/listener/listener.go

```diff
@@ -64,12 +66,26 @@ func (l *StatsDUDPListener) Listen() {
 			level.Error(l.Logger).Log("error", err)
 			return
 		}
-		l.HandlePacket(buf[0:n])
+		// avoid making copies of slices since we need to minimize the time spent here in order not to drop packets
+		l.EnqueueUdpPacket(buf, n)
 	}
 }
 
-func (l *StatsDUDPListener) HandlePacket(packet []byte) {
+func (l *StatsDUDPListener) EnqueueUdpPacket(packet []byte, n int) {
 	l.UDPPackets.Inc()
+	packetCopy := make([]byte, n)
+	copy(packetCopy, packet)
+	l.UdpPacketQueue <- packetCopy
```

Comment:
What happens when the queue is full? Do we want to back pressure here, or would it be useful to drop the packet and increment a counter when this happens? For historical context, we [had this idea before](https://github.com/prometheus/statsd_exporter/pull/196#discussion_r274183349). The upside would be that it's easier to keep track of the dropped packets that way, instead of having to determine it from the kernel. At the time, we ended up not doing that in favor of other internal performance improvements.

Since it seems that you are still running into drops, I was wondering if this would be a good opportunity to revive that approach, especially since it _looks_ like in your single-processor approach it would be fairly easy to do this here?

Reply:
After your comment, I added a counter for UDP packet drops and incremented it when channel is full. Could you please review it?